engine/src/main/java/org/entando/entando/aps/system/services/page/PageService.java

```diff
@@ -260,19 +259,24 @@ public PageDto updatePage(String pageCode, PageRequest pageRequest) {
             throw new ResourceNotFoundException(null, "page", pageCode);
         }
         this.validateRequest(pageRequest);
-        if (!oldPage.getParentCode().equals(pageRequest.getParentCode())) {
-            BeanPropertyBindingResult bindingResult = new BeanPropertyBindingResult(oldPage, "page");
-            bindingResult.reject(PageValidator.ERRCODE_INVALID_PARENT,
-                    new String[]{oldPage.getParentCode(), pageRequest.getParentCode()}, "page.update.parentcode.invalid");
-            throw new ValidationGenericException(bindingResult);
-        }
+
         if (!oldPage.getGroup().equals(pageRequest.getOwnerGroup())) {
             BeanPropertyBindingResult bindingResult = new BeanPropertyBindingResult(oldPage, "page");
             bindingResult.reject(PageValidator.ERRCODE_GROUP_MISMATCH,
                     new String[]{oldPage.getGroup(), pageRequest.getOwnerGroup()}, "page.update.group.invalid");
             throw new ValidationGenericException(bindingResult);
         }
         try {
+            if (!oldPage.getParentCode().equals(pageRequest.getParentCode())) {
```

Comment:
This logic has to exist elsewhere as well for the page move on the page tree. Can we combine or call it once?

Reply:
I searched but I have not found any other check like this in the code. I added this check here to be sure that the user is moving the page from is previous parent branch to a new one.

Reply:
How does it work on the page tree? I'm not talking about the check but the code inside of the block. The calls for set/get seem like they should be elsewhere.

Reply:
Hey @joewhite101 , thanks for the review. 
This code is not duplicated and is present only here (I wrote this from scratch). In this section I'm setting a PagePositionRequest object (to pass it to the movePage method) using the request parameters and calculating the position of the page in the selected branch of the tree. In the "page tree" this object is passed directly from the controller to the movePage method.